cmk/base/plugins/agent_based/checkpoint_connections.py

```diff
@@ -53,14 +64,33 @@ def check_checkpoint_connections(
     params,
     section: Section,
 ) -> CheckResult:
+
+    if section.maximum > 0 and "relative_levels" in params:
+        warn_pct, crit_pct = params["relative_levels"]
+        warn = section.maximum * warn_pct / 100
+        crit = section.maximum * crit_pct / 100
+        levels_upper = (warn, crit)
+    else:
+        # use absolute levels if no relative levels provided or no maximum set on CP
+        levels_upper = params["levels"]
     yield from check_levels(
         value=section.current,
-        levels_upper=params["levels"],
+        levels_upper=levels_upper,
         metric_name="connections",
         label="Current connections",
         render_func=str,
     )
 
+    yield Result(state=State.OK, summary="Peak: {0.peak}".format(section))
+    yield Metric(name="peak", value=section.peak, boundaries=(0, None))
+
+    if section.maximum > 0:
```

Comment:
I think it would make more sense if this part came directly after `check_levels(value=section.current, ...)`, s.t. the results concerning the peak come last